tests/test_scheduler_fn.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Scheduler function tests."""
+import unittest
+from unittest.mock import Mock
+from datetime import datetime
+from flask import Request, Flask
+from werkzeug.test import EnvironBuilder
+from firebase_functions import scheduler_fn
+
+
+class TestScheduler(unittest.TestCase):
+    """
+    Scheduler function tests.
+    """
+
+    def test_on_schedule_decorator(self):
+        """
+        Tests the on_schedule decorator functionality by checking
+        that the __firebase_endpoint__ attribute is set properly.
+        """
+
+        schedule = "* * * * *"
+        tz = "America/Los_Angeles"
+        example_func = Mock(__name__="example_func")
+        decorated_func = scheduler_fn.on_schedule(
+            schedule="* * * * *",
+            timezone=scheduler_fn.Timezone(tz))(example_func)
+        endpoint = getattr(decorated_func, "__firebase_endpoint__")
+
+        self.assertIsNotNone(endpoint)
+        self.assertIsNotNone(endpoint.scheduleTrigger)
+        self.assertEqual(endpoint.scheduleTrigger.get("schedule"), schedule)
+        self.assertEqual(endpoint.scheduleTrigger.get("timeZone"), tz)
+
+    def test_on_schedule_call(self):
+        """
+        Tests to ensure the decorated function is called correctly
+        with appropriate ScheduledEvent object and returns a 200
+        status code if successful.
+        """
+
+        with Flask(__name__).test_request_context("/"):
+            environ = EnvironBuilder(
+                headers={
+                    "X-CloudScheduler-JobName": "example-job",
+                    "X-CloudScheduler-ScheduleTime": "2023-04-13T12:00:00-07:00"
```

Comment:
nit* Thin we are missing a `Z` at the end.

Reply:
Original code expected the Z in the test date time, in prod the dates were coming through without this and weren't an ISO format so this different parsing was needed; here's a log from prod from previous implementation showing the format: 

```
ValueError: time data '2023-04-13T10:19:00-07:00' does not match format '%Y-%m-%dT%H:%M:%SZ'

at ._strptime ( /layers/google.python.runtime/python/lib/python3.10/_strptime.py:349 )
at ._strptime_datetime ( /layers/google.python.runtime/python/lib/python3.10/_strptime.py:568 )
at .on_schedule_wrapped ( /layers/google.python.pip/pip/lib/python3.10/site-packages/firebase_functions/scheduler_fn.py:102 )
at .view_func ( /layers/google.python.pip/pip/lib/python3.10/site-packages/functions_framework/__init__.py:130 )
```

I updated the tests after to match the same format as prod